test/workers.ts

```diff
@@ -0,0 +1,56 @@
+import { resolve } from 'node:path';
+
+import { test } from 'tap';
+
+import Piscina from '../dist';
+
+test('workers are marked as destroyed if destroyed', async t => {
+  let index = 0;
+  // Its expected to have one task get balanced twice due to the load balancer distribution
+  // first task enters, its distributed; second is enqueued, once first is done, second is distributed and normalizes
+  t.plan(4);
+  let workersFirstRound = [];
+  let workersSecondRound = [];
+  const pool = new Piscina({
+    filename: resolve(__dirname, 'fixtures/eval.js'),
+    minThreads: 2,
+    maxThreads: 2,
+    concurrentTasksPerWorker: 1,
+    loadBalancer (_task, workers) {
+      if (workersFirstRound.length === 0) {
+        workersFirstRound = workers;
+        workersSecondRound = workers;
+      } else if (
+        workersFirstRound[0].id !== workers[0].id
+      ) {
+        workersSecondRound = workers;
+      }
+      // Verify distribution to properly test this feature
+      const candidate = workers[index++ % workers.length];
+
+      if (candidate.currentUsage !== 0 && !candidate.isRunningAbortableTask) {
```

Comment:
It's not the purpose of the load balancer to test if the worker chosen is usable or not: it should be factored out. Otherwise LBs will duplicate that code again and again: the main code ask for a worker to be chosen to it, that's all. If the worker is not eligible (busy, not available, aborted), the main code ask again. 
That does not hinder the LB to also decide integrate availability checks, but minimum LB requirement code is to chose a worker.

Reply:
Get your point, but what are you referring to exactly on this?

To bring context, this was boilerplate to test the workers were hinted correctly that the workers were marked as destroyed correctly; true I copy-pasted from other code (that's why agree with your statement).

Reply:
Isn't the conditions testing for the eligibility of the chosen worker? If not, that review comment is wrong. 

The intent is the same as for the test comment: once LB has chosen a worker, the worker eligibility check and/or worker creation conditions check should be done outside the LB code.

Reply:
>Isn't the conditions testing for the eligibility of the chosen worker?

Not really, it just aims to test the workers are marked as `destroyed` whenever they get destroyed; used that _distribution implementation_ for simplicity and copy paste from others